packages/calendar/src/Calendar.ts

```diff
@@ -0,0 +1,449 @@
+                    title="Previous"
+                    class="spectrum-ActionButton spectrum-Calendar-prevMonth"
+                    ?disabled=${this.disabled}
+                    @click=${this.handlePreviousMonth}
+                >
+                    <div slot="icon">
+                        <slot name="prev-icon">
+                            <sp-icon-chevron-left></sp-icon-chevron-left>
+                        </slot>
+                    </div>
+                </sp-action-button>
+
+                <!-- TODO: Translate "Next" -->
+                <sp-action-button
+                    quiet
+                    size="s"
+                    aria-label="Next"
+                    title="Next"
+                    class="spectrum-ActionButton spectrum-Calendar-nextMonth"
+                    ?disabled=${this.disabled}
+                    @click=${this.handleNextMonth}
+                >
+                    <div slot="icon">
+                        <slot name="prev-icon">
+                            <sp-icon-chevron-right></sp-icon-chevron-right>
+                        </slot>
+                    </div>
+                </sp-action-button>
+            </div>
+        `;
+    }
+
+    public renderCalendarGrid(): TemplateResult {
+        return html`
+            <!-- TODO: Implement keyboard navigation -->
+            <div
+                class="spectrum-Calendar-body"
+                role="grid"
+                tabindex=${ifDefined(!this.disabled ? '0' : undefined)}
+                aria-readonly="true"
+                aria-disabled=${this.disabled}
+            >
+                <table role="presentation" class="spectrum-Calendar-table">
+                    ${this.renderCalendarTableHead()}
+                    ${this.renderCalendarTableBody()}
+                </table>
+            </div>
+        `;
+    }
+
+    public renderCalendarTableHead(): TemplateResult {
+        return html`
+            <thead role="presentation">
+                <tr role="row">
+                    ${this._getWeekdays().map((weekday) =>
+                        this.renderWeekdayColumn(weekday)
+                    )}
+                </tr>
+            </thead>
+        `;
+    }
+
+    public renderWeekdayColumn(weekday: CalendarWeekday): TemplateResult {
+        return html`
+            <th
+                role="columnheader"
+                scope="col"
+                class="spectrum-Calendar-tableCell"
+            >
+                <abbr class="spectrum-Calendar-dayOfWeek" title=${weekday.long}>
+                    ${weekday.narrow}
+                </abbr>
+            </th>
+        `;
+    }
+
+    public renderCalendarTableBody(): TemplateResult {
+        return html`
+            <tbody role="presentation">
+                ${[...new Array(this._getWeeksInCurrentMonth()).keys()].map(
+                    (weekIndex) => this.renderCalendarTableRow(weekIndex)
+                )}
```

Comment:
`new Array` here will likely thrash the render. Can you confirm?

If so, we should memoize these keys so that the days get rendered less often.

Reply:
I created a private property to store this information